inc_internal/metrics.h

```diff
@@ -40,28 +41,22 @@ struct rate_s {
     atomic_llong param;
 
     void (*tick_fn)(struct rate_s *);
+    double (*rate_get_fn)(struct rate_s *); //a function that will return the rate as a double
```

Comment:
i don't think you need this func:
* just use `instant_rate()` in `tick_instant` and store it
* regular `metric_get_rate()` calc would work

Reply:
`instant_rate` divides by nanos not seconds. tick_instant does the division ahead of time. i could multiply it by nanos in `tick_instant` if that's what you want? i could then remove the functions stuff i added. this way seemed better than that but i don't care terribly if you want me to rm the functions

Reply:
looks like i could set intervalNanos to 5 - but that also felt awkward. i don't see it used anywhere else though so maybe i'll jsut rename that variable and use it

Reply:
i tried it out... metrics_rate_get also multiplies by `SECOND` which causes another, different wrinkle - i think the way i did it here is the most clear so far

Reply:
I pretty sure you'd get the same result: interval is 5 seconds, intervalNanos is interval in nanos, metric_get_rate scales is it up to rate-per-second. metric->rate would be stored in value/ns and consistent with the way other rates are handled

Reply:
> looks like i could set intervalNanos to 5 - but that also felt awkward. i don't see it used anywhere else though so maybe i'll jsut rename that variable and use it

`intervalNanos` is used in `instant_rate()`

Reply:
i've removed the functions at your request